cmd/kubelet/app/server.go

```diff
@@ -425,7 +425,9 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 			if err != nil {
 				return err
 			}
-			if cloud == nil {
+			if strings.EqualFold(s.CloudProvider, cloudprovider.ExternalCloudProvider) {
```

Comment:
EqualFold feels like overkill.  We don't really want to tell people that "ExTerNal" is just as good, because someone somewhere will just use `==`.  Alternately, do an IsExternal(string) function and don't publish the const.